backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/utils/VdsCpuUnitPinningHelper.java

```diff
@@ -224,4 +232,44 @@ private int getAvailableCores(List<VdsCpuUnit> cpuTopology, int socket, int vThr
         }
         return count;
     }
+
+    /**
+     * Filters and removes physical sockets from being available based on the VM NUMA nodes memory requirements.
+     * The memory is accumulated per physical socket (can be multiple physical NUMA nodes), as we use INTERLEAVED
+     * tune for the pinning.
+     * The cpuTopology is a deep copy made locally for this specific call, therefore we can change it.
+     * @param cpuTopology List<{@link VdsCpuUnit}>. The list of VdsCpuUnit.
+     * @param vm VM object.
+     * @param vdsId GUID of the VDS object.
+     */
+    private void filterBadNumaMemorySockets(List<VdsCpuUnit> cpuTopology, VM vm, Guid vdsId) {
+        if (vm.getCpuPinningPolicy() != CpuPinningPolicy.DEDICATED) {
+            return;
+        }
+
+        List<VdsNumaNode> vdsNumaNodes = vdsNumaNodeDao.getAllVdsNumaNodeByVdsId(vdsId);
+        if (vdsNumaNodes == null || vdsNumaNodes.isEmpty()) {
+            return;
+        }
+
+        Long highestMemRequired = Collections.max(vm.getvNumaNodeList().stream().map(NumaNode::getMemTotal).collect(Collectors.toList()));
+        for (int socket : getOnlineSockets(cpuTopology)) {
+            List<Integer> numaIdsInSocket = getNumaIdsInSocket(cpuTopology, socket);
+            List<VdsNumaNode> numasInSocket = vdsNumaNodes.stream()
+                    .filter(numa -> numaIdsInSocket.contains(numa.getIndex()))
+                    .collect(Collectors.toList());
+            Long totalMemory = numasInSocket.stream()
+                    .map(VdsNumaNode::getNumaNodeStatistics)
+                    .map(NumaNodeStatistics::getMemFree)
+                    .reduce(0L, Long::sum);
+            if (totalMemory < highestMemRequired) {
+                // Memory is missing or there is no enough total memory available for the socket.
+                cpuTopology.removeAll(cpuTopology.stream().filter(cpu -> cpu.getSocket() == socket).collect(Collectors.toList()));
+            }
+        }
```

Comment:
we talked about moving this to the CPU placement to take into consideration placement of two vnuma nodes within the same psocket

Reply:
ok, that was an initial filtering - so it didn't take that into consideration